src/components/SearchField.tsx

```diff
@@ -1,38 +1,241 @@
+    },
+    [checkboxes]
+  );
+
+  const selectGenreHandler = useCallback(
+    (event: React.ChangeEvent<HTMLSelectElement>): void => {
+      setGenreToSearch(event.target.value);
+    },
+    []
+  );
 
   return (
     <>
-      <form onSubmit={submitHandler} className="search-form">
+      <form onSubmit={submitSearchRequest} className="search-form">
         <input
+          onBlur={onBlurSearchInput}
           value={inputValue}
-          onChange={changeHandler}
+          onChange={searchInputChange}
           id="search"
           type="search"
           placeholder="Search"
           className="header-search"
+          onFocus={onFocusSearchInput}
+          autoComplete="off"
         />
-
-        <button type="submit">
+        {(focus || dropdown) && (
+          <div
+            className="dropdown-filter"
+            onMouseEnter={mouseEnterDropdown}
+            onMouseLeave={mouseLeaveDropdown}>
+            <div className="dropdown-wrapper">
+              <div className="date-select-wrapper">
+                <span>
+                  <button onClick={fromDateInputDecrease}>-</button>
+                  <input
+                    type="number"
+                    min="1980"
+                    value={dateRange.fromDate}
+                    readOnly
+                  />
+                  <button onClick={fromDateInputIncrease}>+</button>
+                </span>
+                <span>
+                  <button onClick={toDateInputDecrease}>-</button>
+                  <input type="number" value={dateRange.toDate} readOnly />
+                  <button onClick={toDateInputIncrease}>+</button>
+                </span>
+              </div>
+              <label
+                className={`dropdown-item ${
+                  checkboxes.favorites ? 'active-checkbox' : ''
+                }`}>
+                <input
+                  onChange={favoritesCheckboxHandler}
+                  type="checkbox"
+                  className="custom-checkbox"
+                  checked={checkboxes.favorites}
+                />
+                <span>Favorite</span>
+                <img
+                  className="check-img"
+                  src={checkboxes.favorites ? checkedImg : uncheckedImg}
+                  alt=""
+                />
+              </label>
+              <label
+                className={`dropdown-item ${
+                  checkboxes.history ? 'active-checkbox' : ''
+                }`}>
+                <input
+                  onChange={historyCheckboxHandler}
+                  type="checkbox"
+                  className="custom-checkbox"
+                  checked={checkboxes.history}
+                />
+                <span>History</span>
+                <img
+                  className="check-img"
+                  src={checkboxes.history ? checkedImg : uncheckedImg}
+                  alt=""
+                />
+              </label>
+              <select
```

Comment:
This select does not look like in the design, but maybe it is not crucial..
But in collapsed state its height is lower than other inputs:

![image](https://user-images.githubusercontent.com/35143261/114882047-358a8a00-9e0c-11eb-8da1-1cad2ef0d53d.png)

And also without any label for input it is not clear what filter is set as None by default. The text for this default value could be something like "Select genre" or what ever you prefer, to give user understanding what criteria for search in this select

Reply:
fixed it